decider-bcl2fastq/src/main/java/ca/on/oicr/pde/deciders/Bcl2fastqDecider.java

```diff
@@ -591,31 +627,82 @@ public void setConfig(Map<String, String> config) {
+                    .map(barcodeString -> {
+                        try {
+                            return Barcode.fromString(barcodeString);
+                        } catch (DataMismatchException ex) {
+                            unsupportedBarcodeStrings.add(barcodeString);
+                            return null;
+                        }
+                    })
+                    .collect(Collectors.toList());
+            if (!unsupportedBarcodeStrings.isEmpty()) {
+                addInvalidLane("Error while generating workflow run(s) for lane = [{0}], unsupported barcode(s):\n{1}", laneName, unsupportedBarcodeStrings.toString());
+                continue;
+            }
+
+            //check for barcode collisions
+            List<BarcodeCollision> collisions = BarcodeComparison.getTruncatedHammingDistanceCollisions(laneBarcodes, minAllowedEditDistance);
+            if (collisions.size() > 0) {
+                addInvalidLane("Error while generating workflow run(s) for lane = [{0}], barcode collision(s):\n{1}", laneName, Joiner.on("\n").join(collisions));
+                continue;
+            }
+
+            //more than one sample in the lane, demultiplexing must be done for all workflow runs scheduled for this lane
+            boolean doDemulitplexing = getIsDemultiplexSingleSampleMode();
+            if (samples.size() > 1) {
+                doDemulitplexing = true;
+            }
+
+            //group lane samples by barcode length
+            Map<String, List<ProvenanceWithProvider<SampleProvenance>>> samplesGroupedByBarcodeLength = samples.stream()
+                    .collect(Collectors.groupingBy(s -> {
+                        try {
+                            return Barcode.fromString(s.getProvenance().getIusTag()).getLengthString();
+                        } catch (DataMismatchException ex) {
+                            //this error should have been handled already so this exception should not be thrown
+                            throw new RuntimeException(ex);
+                        }
+                    }));
+
+            //iterate over all lane samples grouped by barcode length
+            try {
+                validWorkflowRuns.addAll(generateWorkflowRunsForLane(Iterables.getOnlyElement(laneNameToLaneProvenance.get(laneName)),
+                        samplesGroupedByBarcodeLength, handler, doDemulitplexing));
+            } catch (InvalidLaneException ex) {
+                addInvalidLane("Error while generating workflow run(s) for lane = [{0}]:\n{1}", laneName, ex.toString());
             }
         }
 
-        //list invalid workflow runs
-        for (WorkflowRun wr : invalidWorkflowRuns) {
-            log.info("Invalid workflow run:\n" + debugWorkflowRun(wr));
+        //list invalid lanes
+        for (String invalidLane : this.laneErrors) {
+            log.error(invalidLane);
+//            log.warn("Invalid workflow run:\n" + debugWorkflowRun(wr));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)